src/misc/lv_event.c

```diff
@@ -222,3 +241,44 @@ void lv_event_mark_deleted(void * target)
 /**********************
  *   STATIC FUNCTIONS
  **********************/
+
+static void cleanup_event_list(lv_event_list_t * list)
+{
+    if(list->is_traversing) return;
+    if(list->has_marked_deleting == false) return;
+
+    uint32_t size = event_array_size(list);
+    uint32_t kept_count = 0;
+    for(uint32_t i = 0; i < size; i++) {
+        lv_event_dsc_t ** dsc_i = event_array_at(list, i);
+        lv_event_dsc_t ** dsc_kept = event_array_at(list, kept_count);
+        if(event_is_marked_deleting(*dsc_i) == false) {
+            *dsc_kept = *dsc_i;
+            kept_count++;
+        }
+        else lv_free(*dsc_i);
+    }
+
+    if(kept_count == 0) lv_array_deinit(&list->array);
+    else lv_array_resize(&list->array, kept_count);
```

Comment:
lv_array_resize(&list->array, count);

Reply:
```cpp
void lv_array_resize(lv_array_t * array, uint32_t new_capacity)
{
    uint8_t * data = lv_realloc(array->data, new_capacity * array->element_size);
    LV_ASSERT_NULL(data);
    array->data = data;
    array->capacity = new_capacity;
```

it's an UB here assuming realloc(ptr, 0) will free the ptr